### PR TITLE
[5.1] Validate Password Reset Token On Reset Password Page

### PR DESCRIPTION
**Pull request by razamehdi-vteams**
Check Validity of The Password Reset Token when the user visits the Password Reset page.

By default, password reset is always shown regardless of whether the reset token is valid/invalid. In this pull request, i have modified the flow as follows:
- Checks Whether The Reset Token Exists in the Database.
- If token is found, then token expiration datetime is calculated. If expiration datetime is greater than the current datetime, then token is considered valid. Otherwise the token is invalid.

### CODE REVIEW

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell This is a weird issue, i am getting error StyleCI error for a file which i didnt edit.

**Comment by GrahamCampbell** [comment]
@razamehdi-vteams You need to rebase.

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell After rebase, should i generate another pull request?

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell rebase completed. Kindly check & confirm.

**Comment by GrahamCampbell** [comment]
That's not a rebase. You need to rebase without any commits.

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell I used this link for rebasing 

https://help.github.com/articles/configuring-a-remote-for-a-fork/
https://help.github.com/articles/syncing-a-fork/

I followed the steps in above links to rebase. This removed the StyleCI issue i was seeing with this pull request.

**Comment by GrahamCampbell** [comment]
Now there are 5 commits.

**Comment by GrahamCampbell** [comment]
Are you forgetting to force push?

File: src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php [hunk 1]
```diff
@@ -172,6 +172,28 @@ public function createNewToken()
     }
 
     /**
+     * Get and validate token data from database.
+     *
+     * @param string $token
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php]: cs

File: src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php [hunk 2]
```diff
@@ -37,4 +37,12 @@ public function delete($token);
      * @return void
      */
     public function deleteExpired();
+
+    /**
+     * Get and validate token data from database.
+     *
+     * @param string $token
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php]: cs

File: src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php [hunk 3]
```diff
@@ -37,4 +37,12 @@ public function delete($token);
...
+     * @return bool
+     */
+    public function validateToken($token);
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php]: adding a method to an interface is a breaking change

File: src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php [hunk 4]
```diff
@@ -172,6 +172,28 @@ public function createNewToken()
...
+     */
+    public function validateToken($token)
+    {
+        $token = $this->getTable()->where('token', $token)->first();
+
+        if (empty($token)) {
+            return false;
+        }
+
+        $expiration_date = Carbon::parse($token->created_at)->addSeconds($this->expires);
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php]: you MUST use camelCase

Comment (razamehdi-vteams) [inline src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php]: So the name should be changed from *validateToken* to *ValidateToken*?

Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php]: no

Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php]: expiration_date must be expirationDate

**Comment by KennedyTedesco** [comment]
There is no need of get all fields from the table. Also, maybe just this is enough:

```php
/**
 * Get and validate token data from database.
 *
 * @param string $token
 * @return bool
 */
public function validateToken($token)
{
    $token = $this->getTable()->where('token', $token)->first(['created_at']);
    return $token && Carbon::parse($token->created_at)->addSeconds($this->expires)->lt(Carbon::now());
}
```

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell Is the latest commit correct?

**Comment by GrahamCampbell** [comment]
No, and you have a merge commit there.

**Comment by GrahamCampbell** [comment]
Did you forget to force push?

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell no i didnt forget to force push.

**Comment by GrahamCampbell** [comment]
Could you fix the cs issues i've commented, squash to one commit and force push please.

**Comment by GrahamCampbell** [comment]
There should be zero merge commits.

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell you mean i need to fix CS issues to make them PSR-2 compliant correct?

**Comment by GrahamCampbell** [comment]
> you mean i need to fix CS issues to make them PSR-2 compliant correct?

No, to make the phpdoc match laravel's standards.

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell Please check & confirm.

**Comment by GrahamCampbell** [comment]
You have 8 commits. There should only be 1.

**Pull request closed without merging**